SpecialContentStaging.php

```diff
@@ -67,30 +89,48 @@ function execute( $par ) {
 						if ( $stage !== "production" ) $target = $keys[$element + 1];
 
 						$currPage = $stages[$stage];
-						$pageNextStage = null;
-						if ( !empty( $target ) ) {
-							$pageNextStage = $stages[$target];
-						}
+						$pageNextStage = $stages[$target];
+
 						$stagingStatus = "<span style=\"color: green\">&#10003;</span>";
 
-						if ( $stage !== "production" && ( !$pageNextStage || $this->replaceStageInternalRefs( $this->pagePrefix, $currPage->getText(), $source, $target ) !== $pageNextStage->getText() ) ) {
-							$stagingStatus = "<html><a href=\"?title=Special:ContentStaging&prefix=" . $this->pagePrefix .
-									"&action=copy&page=" . $currPage->getId() .
-									"&source=" . $source . "&target=" . $target .
-									"\" style=\"color: red;\">&#10007;</a></html>";
+						if( get_class( $currPage ) !== 'WikiPage' ) {
```

Comment:
It is partly caused by the structure of `$this->stages` but I find this line to be rather tricky way to check if there is a page existing for the given page and stage.